util/eventbus/client.go

```diff
@@ -147,6 +147,26 @@ func Subscribe[T any](c *Client) *Subscriber[T] {
 	return s
 }
 
+// SubscribeFunc is like [Subscribe], but calls the provided func for each
+// event of type T.
+//
+// A SubscriberFunc calls f synchronously from the client's goroutine.
```

Comment:
I suggest amplifying this sentence a bit: f must not block for any length of time, because doing so risks locking up the entire event bus, or causing significant hiccups.

As phrased right now, I could be excused for going "oh cool, I have a goroutine all to myself and can do what I like with it", whereas we intend to convey "you're borrowing time on a goroutine that needs to keep up with incoming events or Bad Things will happen, be mindful of how much work you do here".

This is the same contract as the channel-based event API, so nothing new per se, but I think it's worth reinforcing in the user's mind, because it's one of the major design points that affects how you interact with the facility.

Reply:
Coda: or perhaps the package-wide documentation already makes this contract clear enough that it doesn't need repeating here? I defer to y'all with your more holistic and fresh view of the package as a whole.

I think I'm indexing on this particularly because it's the one big tradeoff in the bus's design that I'm not super happy with (I'm not advocating for revisiting, the alternatives all have worse side-effects), in the sense that it's a way of holding it wrong that we can't protect the user from. So, I feel a bit of a need to gently reinforce that "don't block" contract frequently, if that makes sense.

Reply:
On the other hand, if we don't mind an extra gorouine, there is a simpler approach, cf 3e2be380. Much less disruptive, but a fresh goroutine per event, which maybe we do not want. I offer it as an alternative to deeper changes.